APIMatic.Core/APIMatic.Core.csproj

```diff
@@ -29,6 +30,7 @@
     </ItemGroup>
 
     <ItemGroup>
+        <PackageReference Include="Microsoft.Bcl.AsyncInterfaces" Version="8.0.1" />
```

Comment:
You might have to resolve the warning related to this new dependency in both Core and Core.Test projects
![image](https://github.com/user-attachments/assets/ae053040-773e-4f19-80b1-c304203805ef)

Can we also update Core.Test project to net9.0?